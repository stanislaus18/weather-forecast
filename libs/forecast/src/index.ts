export * from './lib/forecast.module';
export * from './lib/forecast.component';
