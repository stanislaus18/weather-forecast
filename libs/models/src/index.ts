export * from './lib/weather';
export * from './lib/common';