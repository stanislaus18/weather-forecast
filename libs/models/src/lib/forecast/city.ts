import { Coord } from '../common'

export interface City {
    id: number
    name: string
    coord: Coord
    country: string
    population: number
    timezone: number
    sunrise: number
    sunset: number
}
