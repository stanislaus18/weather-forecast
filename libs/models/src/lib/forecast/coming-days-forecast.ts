
export class ComingDaysForecast {
  temperatureMax: number;
  temperatureMin: number;
  date: string;

  constructor(temperatureMax: number, temperatureMin: number, date: string) {
    this.temperatureMax = temperatureMax;
    this.temperatureMin = temperatureMin;
    this.date = date;
  }
}

