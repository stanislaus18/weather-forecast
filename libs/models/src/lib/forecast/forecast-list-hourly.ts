
export interface ForecastListHourly {
  temperature_2m: number[];
  time: string[];
  weathercode: number[];
}