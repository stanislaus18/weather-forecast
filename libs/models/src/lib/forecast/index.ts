export * from './forecast';
export * from './forecast-state';
export * from './forecast-list';
export * from './forecast-state-list';