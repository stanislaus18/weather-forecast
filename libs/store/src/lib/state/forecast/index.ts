export * from './forecast.actions';
export * from './forecast.state';
export * from './forecast.state.module';
