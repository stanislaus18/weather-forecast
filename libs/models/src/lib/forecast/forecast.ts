import { City } from './city';
import { ForecastList } from './forecast-list';

export interface Forecast {
    cod: string
    message: number
    cnt: number
    list: ForecastList[]
    city: City
  }

  
  