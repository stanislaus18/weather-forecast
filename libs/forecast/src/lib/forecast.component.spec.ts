import { TestBed } from '@angular/core/testing';

import { provideAutoSpy, Spy } from 'jest-auto-spies';

import { ForecastFacadeService } from '@weather-forecast/store';
import { ComingDaysForecast, Forecast } from '@weather-forecast/models';

import { ForecastComponent } from './forecast.component';

describe('ForecastComponent', () => {
  let component: ForecastComponent;
  let forecastFacadeServiceSpy: Spy<ForecastFacadeService>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [
        ForecastComponent,
        provideAutoSpy(ForecastFacadeService, {
          observablePropsToSpyOn: ['todaysForecast$', 'comingDaysForecast$'],
        })
      ],
    });

    component = TestBed.inject(ForecastComponent);
    forecastFacadeServiceSpy =  TestBed.inject(ForecastFacadeService) as Spy<ForecastFacadeService>;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('loadTodaysVisualization with appropriate values', done => {
    // arrange
    const mockData = [{
      time: '12:00',
      temperature: 21.01,
      description: 'clear sky'
    },
    {
      time: '13:00',
      temperature: 19.01,
      description: 'outcast'
    }];

    forecastFacadeServiceSpy.todaysForecast$.nextOneTimeWith(mockData as Forecast[])

    // act 
    component.ngOnInit();

    component.todaysForecast$?.subscribe(() => {
      // assert
      expect(component.chartOptionToday).toMatchSnapshot();
      done();
    });
  });

  it('loadComingDaysVisualization with appropriate values', done => {
    // arrange
    const mockData = [{
      date: '2023-08-21',
      temperatureMax: 21.01,
      temperatureMin: 19.1
    },
    {
      date: '2023-08-21',
      temperatureMax: 19.01,
      temperatureMin: 17.9
    }];

    forecastFacadeServiceSpy.comingDaysForecast$.nextOneTimeWith(mockData as ComingDaysForecast[])

    // act 
    component.ngOnInit();

    component.comingDaysForecast$?.subscribe(() => {
      // assert
      expect(component.chartOptionComingDays).toMatchSnapshot();
      done();
    });
  });
});
