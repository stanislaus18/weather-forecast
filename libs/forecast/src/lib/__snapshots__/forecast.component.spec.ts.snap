// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastComponent loadComingDaysVisualization with appropriate values 1`] = `
{
  "color": [
    "#80FFA5",
    "#00DDFF",
  ],
  "grid": {
    "bottom": "3%",
    "containLabel": true,
    "left": "3%",
    "right": "4%",
  },
  "series": [
    {
      "areaStyle": {
        "color": LinearGradient {
          "colorStops": [
            {
              "color": "rgb(255, 0, 135)",
              "offset": 0,
            },
            {
              "color": "rgb(135, 0, 157)",
              "offset": 1,
            },
          ],
          "global": false,
          "type": "linear",
          "x": 0,
          "x2": 0,
          "y": 0,
          "y2": 1,
        },
        "opacity": 0.8,
      },
      "data": [
        19.1,
        17.9,
      ],
      "emphasis": {
        "focus": "series",
      },
      "lineStyle": {
        "width": 0,
      },
      "name": "Temperature Min",
      "showSymbol": false,
      "smooth": true,
      "stack": "Total",
      "type": "line",
    },
    {
      "areaStyle": {
        "color": LinearGradient {
          "colorStops": [
            {
              "color": "rgb(255, 191, 0)",
              "offset": 0,
            },
            {
              "color": "rgb(224, 62, 76)",
              "offset": 1,
            },
          ],
          "global": false,
          "type": "linear",
          "x": 0,
          "x2": 0,
          "y": 0,
          "y2": 1,
        },
        "opacity": 0.8,
      },
      "data": [
        21.01,
        19.01,
      ],
      "emphasis": {
        "focus": "series",
      },
      "label": {
        "position": "top",
        "show": true,
      },
      "lineStyle": {
        "width": 0,
      },
      "name": "Temperature Max",
      "showSymbol": false,
      "smooth": true,
      "stack": "Total",
      "type": "line",
    },
  ],
  "title": {
    "text": "Coming days forecast",
  },
  "tooltip": {
    "axisPointer": {
      "label": {
        "backgroundColor": "#6a7985",
      },
      "type": "cross",
    },
    "trigger": "axis",
  },
  "xAxis": [
    {
      "boundaryGap": false,
      "data": [
        "2023-08-21",
        "2023-08-21",
      ],
      "type": "category",
    },
  ],
  "yAxis": [
    {
      "type": "value",
    },
  ],
}
`;

exports[`ForecastComponent loadTodaysVisualization with appropriate values 1`] = `
{
  "grid": {
    "bottom": "3%",
    "containLabel": true,
    "left": "3%",
    "right": "4%",
  },
  "legend": {
    "data": [
      "Temperature",
    ],
  },
  "series": [
    {
      "areaStyle": {
        "color": LinearGradient {
          "colorStops": [
            {
              "color": "rgb(224, 62, 76)",
              "offset": 0,
            },
            {
              "color": "rgb(255, 191, 0)",
              "offset": 1,
            },
          ],
          "global": false,
          "type": "linear",
          "x": 0,
          "x2": 0,
          "y": 0,
          "y2": 1,
        },
        "opacity": 0.8,
      },
      "data": [
        21.01,
        19.01,
      ],
      "emphasis": {
        "focus": "series",
      },
      "label": {
        "position": "top",
        "show": true,
      },
      "name": "Temperature",
      "stack": "Total",
      "type": "line",
    },
  ],
  "title": {
    "text": "Todays forecast",
  },
  "tooltip": {
    "axisPointer": {
      "label": {
        "backgroundColor": "#6a7985",
      },
      "type": "cross",
    },
    "trigger": "axis",
  },
  "xAxis": [
    {
      "boundaryGap": false,
      "data": [
        "12:00",
        "13:00",
      ],
      "type": "category",
    },
  ],
  "yAxis": [
    {
      "splitLine": {
        "show": false,
      },
      "type": "value",
    },
  ],
}
`;
